File: python/test_Table.py
import unittest
from KombiN import Table


class TableTest(unittest.TestCase):

    def test_all_30_false(self):
        result = True
        for X in range(1, 31):
            for Y in range(1, 31):
                _X_Y = Table(X, Y, False)
                for i in range(1, (X * Y) + 1):
                    Ai, Bi = _X_Y.GetElementsAtIndex(i)
                    index = _X_Y.GetIndexOfElements(Ai, Bi)
                    if(i != index):
                        # result = False
                        raise ValueError(f"i:{i}#index{index} for Ai:{Ai}, Bi:{Bi}")
        self.assertEqual(result, True)

    def test_all_30_true(self):
        result = True
        for X in range(1, 31):
            for Y in range(1, 31):
                _X_Y = Table(X, Y, True)
                for i in range(0, (X * Y)):
                    Ai, Bi = _X_Y.GetElementsAtIndex(i)
                    index = _X_Y.GetIndexOfElements(Ai, Bi)
                    if(i != index):
                        # result = False
                        raise ValueError(f"i:{i}#index{index} for Ai:{Ai}, Bi:{Bi}")
        self.assertEqual(result, True)

File: csharp/kombin.tests/TestTable.cs

using System;
using Xunit;

[assembly: CLSCompliant(true)]
namespace Ninja.Pranav.Algorithms.Kombin.tests {

    /// <summary>
    /// Tests for Table class.
    /// </summary>
    public class TestTable {

        /// <summary>
        /// Rigorous Test :-) All 100 False
        /// </summary>
        [Fact]
        public void TestAll100False() {
            bool result = true;
            for (int X = 1; X <= 100; X++) {
                for (int Y = 1; Y <= 100; Y++) {
                    Table myXY = new Table(X, Y, false);
                    for (long i = 1; i <= X * Y; i++) {
                        long ai, bi;
                        (ai, bi) = myXY.GetElementsAtIndex(i);
                        long index = myXY.GetIndexOfElements(ai, bi);
                        if (i != index) {
                            result = false;
                        }
                    }
                }
            }
            Assert.True(result);
        }

        /// <summary>
        /// Rigorous Test :-) All 100 True
        /// </summary>
        [Fact]
        public void TestAll100True() {
            bool result = true;
            for (int X = 1; X <= 100; X++) {
                for (int Y = 1; Y <= 100; Y++) {
                    Table myXY = new Table(X, Y, true);
                    for (long i = 0; i <= (X * Y) - 1; i++) {
                        long ai, bi;
                        (ai, bi) = myXY.GetElementsAtIndex(i);
                        long index = myXY.GetIndexOfElements(ai, bi);
                        if (i != index) {
                            result = false;
                        }
                    }
                }
            }
            Assert.True(result);
        }
    }
}
